coordinatorlib/src/main/scala/com/socrata/datacoordinator/secondary/PlaybackToSecondary.scala

```diff
@@ -302,47 +311,94 @@ class PlaybackToSecondary[CT, CV](u: PlaybackToSecondary.SuperUniverse[CT, CV],
-                val latest = w.latest(datasetInfo)
-                val latestDataVersion = latest.dataVersion
-                val latestLifecycleStage = latest.lifecycleStage
-                for(copy <- allCopies) {
-                  val secondaryDatasetInfo = makeSecondaryDatasetInfo(copy.datasetInfo)
-                  timingReport("copy", "number" -> copy.copyNumber) {
-                    if(copy.lifecycleStage == metadata.LifecycleStage.Discarded) {
-                      currentCookie = secondary.store.dropCopy(secondaryDatasetInfo.internalName,
-                                                               copy.copyNumber,
-                                                               currentCookie)
-                    } else if(copy.lifecycleStage != metadata.LifecycleStage.Unpublished) {
-                      syncCopy(copy)
-                    } else if(secondary.store.wantsWorkingCopies) {
-                      syncCopy(copy)
-                    } else { /* ok */ }
-                  }
+      val latestCopyInfo = retrying[Option[metadata.CopyInfo]]({
+        timingReport("resync", "dataset" -> datasetId) {
+          u.commit()
+          val r = u.datasetMapReader
+          r.datasetInfo(datasetId, repeatableRead = true) match {
+            case Some(datasetInfo) =>
+              val allCopies = r.allCopies(datasetInfo) // guarantied to be ordered by copy number
+            val latest = r.latest(datasetInfo) // this is the newest _living_ copy
+              for (copy <- allCopies) {
+                val secondaryDatasetInfo = makeSecondaryDatasetInfo(copy.datasetInfo)
+                timingReport("copy", "number" -> copy.copyNumber) {
+                  // secondary.store.resync(.) will be called
+                  // on copies in order by their copy number
+                  val isLatestCopy = copy.copyNumber == latest.copyNumber
+                  if (copy.lifecycleStage == metadata.LifecycleStage.Discarded) {
+                    currentCookie = secondary.store.dropCopy(secondaryDatasetInfo.internalName,
+                      copy.copyNumber,
+                      currentCookie)
+                  } else if (copy.lifecycleStage != metadata.LifecycleStage.Unpublished) {
+                    syncCopy(copy, isLatestCopy)
+                  } else if (secondary.store.wantsWorkingCopies) {
+                    syncCopy(copy, isLatestCopy)
+                  } else {  /* ok */ }
                 }
-                updateSecondaryMap(latestDataVersion, latestLifecycleStage)
-              case None =>
-                drop()
-            }
+              }
+              // end transaction to not provoke a serialization error from touching the secondary_manifest table
+              u.commit()
+              Some(latest)
+            case None =>
+              drop()
+              None
           }
-          return
-        } catch {
+        }
+      }, { case e: Throwable =>
+        e match {
```

Comment:
You can do
```scala
{
  case ResyncSecondaryException(reason) => ...
  case _ => ...
}
```

here instead of `{case e: Throwable => e match {...}}`